dagql/objects.go

```diff
@@ -418,6 +418,14 @@ func (r ObjectResult[T]) MarshalJSON() ([]byte, error) {
 	return r.Result.MarshalJSON()
 }
 
+func (r ObjectResult[T]) DerefValue() (AnyResult, bool) {
+	derefableSelf, ok := any(r.self).(DerefableResult)
+	if !ok {
+		return r, true
```

Comment:
To expand on the PR description, this is the crucial line. Previously calling `DerefValue` on an `ObjectResult` went to `Result.DerefValue` because `Result` is embedded in `ObjectResult`. That resulted in us returning `r` as in the embedded `Result` rather than the `r` here, which is `ObjectResult`.